core/domain/exp_services_test.py

```diff
@@ -2638,6 +2638,49 @@ def test_update_written_translations_with_list_fails(self):
                     self.init_state_name, 'written_translations',
                     [1, 2]), 'Added fake text translations.')
 
+    def test_migrate_exploration_to_the_latest_schema_version(self):
```

Comment:
Here & below: great test names describe both the condition being tested and the expected outcome. That format really helps ensure that test names can easily be used to evaluate all of the different behaviors being tested. In this case, these tests indicate the action (migrate_exploration) and the condition (latest/not latest schema version), but not what is expected to happen. I suggest updating the names to include the expected outcomes.

Reply:
I think what's expected to happen is that the exploration state schema is migrated to the latest version...I could add a _migrates_successfully as a suffix? What are your thoughts @BenHenning?

Reply:
Yeah, I think that's the right thought. What do you think for this & the next:

```python
def test_migrate_exp_to_latest_version_migrates_to_version(self)

def test_migrate_exp_to_earlier_version_raises_exception(self)
```

Both are well under the 80 character limit, and hopefully encapsulate the action being tested (exploration migration), the conditions (latest/not latest version), and the expected outcomes (migrates to the correct version/fails with an exception being raised).

Reply:
Thanks for the suggestions Ben :)